plugin/plugin_test.go

```diff
@@ -0,0 +1,127 @@
+package plugin
+
+import (
+	context "context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAeadPluginWrapper(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+
+	pluginPath := os.Getenv("PLUGIN_PATH")
+	if pluginPath == "" {
+		t.Skipf("skipping plugin test as no PLUGIN_PATH specified")
+	}
+
+	wrapper, cleanup := TestPlugin(t, pluginPath)
+
+	require.NotNil(cleanup)
```

Comment:
Instead of nil-checking here and elsewhere whenever `TestPlugin` is called, would it be slightly better for TestPlugin to just return a cleanup func that no-ops if there's nothing to clean up?

Reply:
Sure, good idea.

Reply:
It actually already does that, basically! It can't return with anything to cleanup without it already failing the test. But I'll make it more explicit.